documentation/step-by-step-install.md

```diff
@@ -0,0 +1,148 @@
+# Step by step install steps and example running the python script
+
+## Get the code
+
+<pre>
+git clone https://github.com/CDLUC3/counter-processor.git
+cd counter-processor
+git checkout <em>branch-or-tag</em>
+</pre>
+
+## Set an environment for python besides your system environment
+<pre>
+python3.6 -m venv counter-env
+python --version # should display the python version 3.6.x in response
+</pre>
+
+## Get the GeoLite2 Country database
+<pre>
+cd maxmind_geoip # go to the maxmind directory
+
+## download the database
+wget https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz
+
+tar zxvf GeoLite2-Country.tar.gz # extract the database
+
+# copy it here for the default config to work
+cp GeoLite2-Country_<em>release-date</em>/GeoLite2-Country.mmdb .
+</pre>
+
+## Edit the *config/config.yaml* file
+- to have your correct *platform* name.
+This would be your repository name that is submitted to the Counter hub.
+- modify the regular expressions in the *path_types* section of the config.yaml file
+for classifying urls (or any string patterns) into *investigations* and *requests*.
+  - While the Counter CoP, counts a *request* as an *investigation*, you will want to classify each as a distinct type
+  with your regular expression and not overlap the two. The processor will
+  take care of "double-counting"  all *requests* as an *investigation* for you.
+- *hub_api_token*: set this value in a *secrets.yaml* file in the same directory as your *config.yaml*
+if you are committing your *config.yaml* to a public respository. Be sure to exclude the *secrets.yaml*
+from being committed to a public place such as with a .gitignore file.  The *hub_api_token* is used for
+automatically submitting to the hub if you have that option enabled.
+
+## Ensure that the program's dependencies are installed
+
+<pre>
+# to get back to your counter-processor top-level directory
+cd ..
+
+# be sure pip has the correct requirements installed
+pip3 install -r requirements.txt
+</pre>
+
+## Copy a very simple log file and be sure the counter-processor runs
+
+<pre>
+# copy a small sample log to process
```

Comment:
mkdir log

Reply:
Making the log directory shouldn't be necessary since the log directory isn't excluded using .gitignore, just the *.log files in it.  There is a hidden .keep file in the log directory, which I think is committed to the repository.